typescript/src/agents/experimental/remote/agent.ts

```diff
@@ -60,75 +57,144 @@ export class RemoteAgent extends BaseAgent<RemoteAgentRunInput, RemoteAgentRunOu
     _options: BaseAgentRunOptions,
     context: GetRunContext<this>,
   ): Promise<RemoteAgentRunOutput> {
-    const runner = this.createRunner(context);
-
-    const output = await runner(input);
-
-    const message: Message = new AssistantMessage(output);
+    const inputs = Array.isArray(input.prompt)
+      ? input.prompt.map(this.convertToACPMessage)
+      : [this.convertToACPMessage(input.prompt)];
+
+    const url = new URL(this.input.url);
+    url.pathname += "/runs";
+    const response = await fetch(url, {
+      method: "POST",
+      headers: {
+        "Accept": "text/event-stream",
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        agent_name: this.input.agentName,
+        input: inputs,
+        mode: "stream",
+      }),
+      signal: context.signal,
+    });
 
-    return { message };
-  }
+    if (!response.ok) {
+      throw new AgentError(`HTTP error! status: ${response.status}`, [], {
+        context: { message: await response.text() },
+      });
+    }
+
+    const reader = response.body?.getReader();
+    if (!reader) {
+      throw new AgentError("Agent's response is not valid");
+    }
+
+    const decoder = new TextDecoder();
+    let partialData = "";
+    let eventData: any = null;
+
+    function parseSSEEvent(eventString: string) {
+      const lines = eventString.split("\n");
+      const event: any = {};
+      for (const line of lines) {
+        const [key, ...valueParts] = line.split(":");
+        if (key && valueParts && valueParts.length > 0) {
+          const value = valueParts.join(":").trim();
+          event[key.trim()] = value;
+        }
+      }
+      return event;
+    }
```

Comment:
We have an existing dependency that can consume SSE endpoints.

@ai-zen/node-fetch-event-source